source/common/common/token_bucket_impl.cc

```diff
@@ -11,10 +11,19 @@ TokenBucketImpl::TokenBucketImpl(uint64_t max_tokens, TimeSource& time_source, d
 uint64_t TokenBucketImpl::consume(uint64_t tokens, bool allow_partial) {
   if (tokens_ < max_tokens_) {
     const auto time_now = time_source_.monotonicTime();
-    tokens_ = std::min((std::chrono::duration<double>(time_now - last_fill_).count() * fill_rate_) +
-                           tokens_,
-                       max_tokens_);
-    last_fill_ = time_now;
+
+    const uint64_t new_fill_tokens =
+        std::chrono::duration_cast<std::chrono::microseconds>(time_now - last_fill_).count() *
+            fill_rate_ +
+        residual_tokens_;
+
+    if (new_fill_tokens >= 1000000) {
+      const uint64_t last_clac = new_fill_tokens % 1000000;
+      residual_tokens_ = last_clac;
```

Comment:
nit: residual_tokens_ = new_fill_tokens % 1000000;

Reply:
Sorry, I forgot to check this implementation！I will update this right away！

Reply:
> nit: residual_tokens_ = new_fill_tokens % 1000000;

I have updated it! please CR! thanks!